edx_event_bus_redis/internal/consumer.py

```diff
@@ -153,10 +159,13 @@ def _read_pending_msgs(self) -> Optional[tuple]:
         Read pending messages, if no messages found set check_backlog to False.
         """
         logger.debug("Consuming pending msgs first.")
+        if self.claim_msgs_older_than is not None:
+            self.consumer.autoclaim(self.consumer_name, min_idle_time=self.claim_msgs_older_than, count=1)
         msg_meta = self.consumer.pending(count=1, consumer=self.consumer_name)
         if not msg_meta:
             logger.debug("No more pending messages.")
-            self.check_backlog = False
+            self.has_pending_msgs = False
```

Comment:
Having this side-effect here is a little confusing, it took me a minute to understand where the switch from reading pending messages to new ones was. I'd suggest either renaming the function or moving this out to `_consume_indefinitely` where the rest of the `has_pending_msgs` logic lives.

Reply:
@bmtcril I made this change after I realized that we were only processing pending msgs if `check_backlog` is set to true but in reality check_backlog is used if you want to process all the msgs that were not read by this group. This is already in a function called `_read_pending_msgs` and I have updated the function docs a bit. Let me know if I misunderstood something.

Reply:
I guess my confusion here is that it's not just reading pending messages, but also has the side-effect of deciding that there are no more messages to be read and changing the behavior of the parent method to start reading current messages instead. Looking at the code in the parent function it's not obvious what triggers that change, since the name of this method only indicates a read operation.

Reply:
Got it, that makes sense. Moved the part where we are changing the flags out of `read_pending_msgs`.